src/metabase/automagic_dashboards/core.clj

```diff
@@ -863,13 +859,40 @@
   [field opts]
   (automagic-dashboard (merge (->root field) opts)))
 
-(defn- enhanced-table-stats
-  [table]
-  (let [field-types (->> (db/select [Field :special_type] :table_id (u/get-id table))
-                         (map :special_type))]
-    (assoc table :stats {:num-fields  (count field-types)
-                         :list-like?  (= (count (remove #{:type/PK} field-types)) 1)
-                         :link-table? (every? #{:type/FK :type/PK} field-types)})))
+(defn- enhance-table-stats
+  [tables]
+  (when (not-empty tables)
+    (let [field-count (->> (db/query {:select   [:table_id [:%count.* "count"]]
+                                      :from     [Field]
+                                      :where    [:in :table_id (map u/get-id tables)]
```

Comment:
Is there any filtering happening on the list of candidate tables? 

Specifically, with this are we fetching a list of tables only to then put them in this field, when we could just pass in the database ID?

Reply:
We do use some table fields, but it's unnecessary to fetch the entire table object. Have changed to fetch only what's needed.